third_party/nvidia/lib/Dialect/NVWS/Transforms/LowerAref.cpp

```diff
@@ -281,15 +335,18 @@ void rewriteGetEnterOp(ArefCreateOp arefOp, ArefGetEnterOp op,
   auto views = getSubViews(arefVal, op.getStage(), loc, rewriter);
   assert(views.size() == op.getBuffers().size());
 
-  for (int i = 0; i < arefVal.buffers.size(); ++i)
-    op.getBuffers()[i].replaceAllUsesWith(views[i]);
+  for (auto [oldBuffer, view] : llvm::zip(op.getBuffers(), views)) {
+    oldBuffer.replaceAllUsesWith(view);
+    // Before aref lowering, memdesc_trans consumes an immutable buffer from
+    // a get enter op. After lowering, all buffers are mutable.
+    propagateMutability(view);
```

Comment:
I tried to use `replaceUsesAndPropagateType` here but couldn't make it work, probably because `oldBuffer` is an element of an array. `getUses()` iterator range for it is empty.